core/field.js

```diff
@@ -64,1155 +64,1160 @@ goog.require('Blockly.Gesture');
+   * @package
    */
-  this.fieldGroup_ = null;
+  initView() {
+    this.createBorderRect_();
+    this.createTextElement_();
+  }
 
   /**
-   * The rendered field's SVG border element.
-   * @type {SVGRectElement}
-   * @protected
+   * Initializes the model of the field after it has been installed on a block.
+   * No-op by default.
+   * @package
    */
-  this.borderRect_ = null;
+  initModel() {}
 
   /**
-   * The rendered field's SVG text element.
-   * @type {SVGTextElement}
+   * Create a field border rect element. Not to be overridden by subclasses.
+   * Instead modify the result of the function inside initView, or create a
+   * separate function to call.
    * @protected
    */
-  this.textElement_ = null;
+  createBorderRect_() {
+    this.borderRect_ = dom.createSvgElement(
+        Svg.RECT, {
+          'rx': this.getConstants().FIELD_BORDER_RECT_RADIUS,
+          'ry': this.getConstants().FIELD_BORDER_RECT_RADIUS,
+          'x': 0,
+          'y': 0,
+          'height': this.size_.height,
+          'width': this.size_.width,
+          'class': 'blocklyFieldRect',
+        },
+        this.fieldGroup_);
+  }
 
   /**
-   * The rendered field's text content element.
-   * @type {Text}
+   * Create a field text element. Not to be overridden by subclasses. Instead
+   * modify the result of the function inside initView, or create a separate
+   * function to call.
    * @protected
    */
-  this.textContent_ = null;
+  createTextElement_() {
+    this.textElement_ = dom.createSvgElement(
+        Svg.TEXT, {
+          'class': 'blocklyText',
+        },
+        this.fieldGroup_);
+    if (this.getConstants().FIELD_TEXT_BASELINE_CENTER) {
+      this.textElement_.setAttribute('dominant-baseline', 'central');
+    }
+    this.textContent_ = document.createTextNode('');
+    this.textElement_.appendChild(this.textContent_);
+  }
 
   /**
-   * Mouse down event listener data.
-   * @type {?browserEvents.Data}
-   * @private
+   * Bind events to the field. Can be overridden by subclasses if they need to do
+   * custom input handling.
+   * @protected
    */
-  this.mouseDownWrapper_ = null;
+  bindEvents_() {
+    Tooltip.bindMouseEvents(this.getClickTarget_());
+    this.mouseDownWrapper_ = browserEvents.conditionalBind(
+        this.getClickTarget_(), 'mousedown', this, this.onMouseDown_);
+  }
 
   /**
-   * Constants associated with the source block's renderer.
-   * @type {ConstantProvider}
-   * @protected
+   * Sets the field's value based on the given XML element. Should only be called
+   *     by Blockly.Xml.
+   * @param {!Element} fieldElement The element containing info about the
+   *    field's state.
+   * @package
    */
-  this.constants_ = null;
-
-  opt_config && this.configure_(opt_config);
-  this.setValue(value);
-  opt_validator && this.setValidator(opt_validator);
-};
-
-/**
- * The default value for this field.
- * @type {*}
- * @protected
- */
-Field.prototype.DEFAULT_VALUE = null;
```

Comment:
Wait, looking at this again, wouldn't that make `DEFAULT_VALUE` a property on the class constructor rather than the prototype? I don't think we want that.

Reply:
Per email I sent to the team yesterday, it turns out that the Closure compiler does not yet support class fields (`static` or otherwise), so while I do suggest putting `DEFAULT_VALUE` on the constructor it will have to be done outside the `class` body for the time being.